riscv/processor.cc

```diff
@@ -39,7 +40,7 @@ processor_t::processor_t(const char* isa_str, const char* priv_str,
   log_file(log_file), sout_(sout_.rdbuf()), halt_on_reset(halt_on_reset),
   in_wfi(false), check_triggers_icount(false),
   impl_table(256, false), extension_enable_table(isa.get_extension_table()),
-  last_pc(1), executions(1), TM(cfg->trigger_count)
+  last_pc(1), executions(1), TM(cfg->trigger_count), trace_encoder()
```

Comment:
The default constructor for the `trace_encoder` member object is called by default in the constructor of `processor_t`, adding stuff to the initializer list is done to use a non-default constructor.

So you don't need to add `trace_encoder()` here... the default constructor (which is what you are using) will still be executed.